detectable/src/main/java/com/synopsys/integration/detectable/detectable/Requirements.java

```diff
@@ -106,14 +104,30 @@ public File optionalFile(File directory, String filename, RequirementNotMetActio
         return file;
     }
 
+    public File file(String filename) {
+        return file(environment.getDirectory(), filename);
+    }
+
+    public File file(String filename, FailedDetectableResult missingFileResult) {
+        return file(environment.getDirectory(), filename, missingFileResult);
+    }
+
     public File file(File directory, String filename) {
-        return file(directory, filename, true);
+        return file(directory, filename, true, null);
     }
 
-    public File file(File directory, String filename, boolean isRelevant) {
-        //Only difference between Optional File and Required File is Required populate failure, so if optional 'is not met' we can capture that by setting failure.
+    public File file(File directory, String filename, FailedDetectableResult missingFileResult) {
+        return file(directory, filename, true, missingFileResult);
+    }
+
+    public File file(File directory, String filename, boolean isRelevant, @Nullable FailedDetectableResult missingFileResult) {
```

Comment:
Could this instead take in a functional interface that creates the failed result (rather than take in the result itself)? Something like FailedResultCreator. I think it would reveal the API a bit better and makes the caller not have to actually instantiate unless it's called.

Reply:
Sure thing